### PR TITLE
chore(deps): update to storybook v7

### PR DESCRIPTION
**Pull request by Francois-Esquire**
## Description

Upgrades to the latest version of storybook v7.

## Checklist

* [x] :globe_with_meridians: component demo is up-to-date (`yarn start`)
* [ ] :ok_hand: ~style updates are Garden Designer approved (add the
  designer as a reviewer)~
* [ ] :white_check_mark: ~all component states are represented
  (`.is-hovered`, `.is-focused`, etc.)~
* [ ] :metal: ~renders as expected sans Bedrock (`?bedrock=false`)~
* [ ] :memo: ~tested in Chrome, Firefox, Safari, Edge, and IE11~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,23 +4,24 @@
...
   "scripts": {
     "build": "lerna run build --stream",
-    "build:demo": "build-storybook -s packages/bedrock/dist -o demo",
+    "build:demo": "storybook build -s packages/bedrock/dist -o demo",
     "format": "prettier-package-json --write && lerna run format --stream",
     "postinstall": "husky install && lerna bootstrap",
     "lint": "eslint scripts/*.js && lerna run lint --stream",
     "prepare": "yarn build",
-    "start": "start-storybook -s packages/bedrock/dist -p 6006",
+    "start": "storybook dev -s packages/bedrock/dist -p 6006",
```
Comment (jzempel) [inline package.json:12]: while we're at it, can we add the [no-version-updates](https://github.com/zendeskgarden/react-components/blob/4d7f950f6cfaf99a87fbb4bd66eb1ee3818191bf/package.json#L22) flag (if it still exists)?

File: .babelrc.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      {
+        "targets": {
+          "chrome": 100
+        }
+      }
+    ],
+    "@babel/preset-react"
+  ],
+  "plugins": []
+}
```
Comment (jzempel) [inline .babelrc.json:15]: [nit] can you fix the missing EOF newline?

File: .babelrc.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "sourceType": "unambiguous",
+  "presets": [
+    [
+      "@babel/preset-env",
+      {
+        "targets": {
+          "chrome": 100
+        }
```
Comment (jzempel) [inline .babelrc.json:9]: Shouldn't this be the same as https://github.com/zendeskgarden/react-components/blob/4d7f950f6cfaf99a87fbb4bd66eb1ee3818191bf/babel.config.js#L13?

File: .babelrc.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      }
+    ],
+    "@babel/preset-react"
+  ],
+  "plugins": []
```
Comment (jzempel) [inline .babelrc.json:14]: Is this needed?

File: .storybook/main.js [hunk 5]
```diff
@@ -7,19 +7,28 @@
...
+    }
+  }],
+  webpackFinal: config => {
     config.output.hashFunction = 'xxhash64';
-
     return config;
+  },
+  framework: {
+    name: '@storybook/react-webpack5',
+    options: {}
```
Comment (jzempel) [inline .storybook/main.js:29]: is this needed?

File: .storybook/main.js [hunk 6]
```diff
@@ -7,19 +7,28 @@
...
+  },
+  docs: {
+    autodocs: true
   }
-};
+};
```
Comment (jzempel) [inline .storybook/main.js:34]: [nit] please restore EOF newline

**Review by jzempel — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Francois-Esquire) [reply package.json:12]: It no longer exists according to the [new CLI docs](https://storybook.js.org/docs/react/api/cli-options) unfortunately.

File: .babelrc.json (hunk 3 above)
Comment (Francois-Esquire) [reply .babelrc.json:9]: Correct! Will update.

File: .storybook/main.js (hunk 5 above)
Comment (Francois-Esquire) [reply .storybook/main.js:29]: It is.. removing it results in a failing build.

**Comment by Francois-Esquire** [comment]
> Until we're ready to address holistically, please set `showName=false` for the custom bedrock toolbar item. Ref https://storybook.js.org/docs/react/essentials/toolbars-and-globals

I didn't have success removing the Bedrock toolbar name with `showName: false`..

File: .storybook/preview.js [hunk 7]
```diff
@@ -30,6 +30,7 @@ export const globalTypes = {
     description: 'CSS Bedrock',
     defaultValue: 'enabled',
     toolbar: {
+      showName: false,
```
Comment (jzempel) [inline .storybook/preview.js:33]: Appears this dropped out of the documentation in the last couple of days. Probably in response to this deprecation: https://github.com/storybookjs/storybook/blob/362b3e941f6a96e2fb74e4495feb46657ecebc0e/code/addons/toolbars/src/components/ToolbarMenuList.tsx#L34-L40

Is there something that can be done with `title` (either empty string or `null`) to keep this as icon-only for now? If not, no big deal.

File: package.json (hunk 1 above)
Comment (jzempel) [reply package.json:12]: According to `yarn storybook dev --help` it is still there.

File: .storybook/preview.js (hunk 7 above)
Comment (Francois-Esquire) [reply .storybook/preview.js]: Looks like an empty string did the trick.

File: .storybook/main.js [hunk 8]
```diff
@@ -7,19 +7,37 @@
...
   webpackFinal: (config) => {
     config.output.hashFunction = 'xxhash64';
 
     return config;
-  }
+  },
+  managerHead: (head) => `
+    ${head}
+    <link rel="icon" href="https://zendeskgarden.github.io/favicons/garden/favicon.ico" />
+  `
```
Comment (jzempel) [inline .storybook/main.js:42]: Can we revert and keep this aligned with https://github.com/zendeskgarden/react-containers/tree/main/.storybook

File: .storybook/preview.js [hunk 9]
```diff
@@ -26,10 +26,11 @@ export const decorators = [
...
   bedrock: {
-    name: 'bedrock',
     description: 'CSS Bedrock',
     defaultValue: 'enabled',
     toolbar: {
+      // empty string to omit name from toolbar
+      title: '',
       icon: 'link',
       items: [
         { value: 'enabled', title: 'Bedrock enabled' },
```
Comment (jzempel) [inline .storybook/preview.js:36]: Please compare with https://github.com/zendeskgarden/react-containers/blob/3676836c2463bee4163139299f156bffa7d26a39/.storybook/preview.js#L59-L69 which seems to be working as intended in v7.

File: package.json [hunk 10]
```diff
@@ -4,12 +4,12 @@
...
   "scripts": {
     "build": "lerna run build --stream",
-    "build:demo": "build-storybook -s packages/bedrock/dist -o demo",
+    "build:demo": "storybook build -o demo",
     "format": "prettier-package-json --write && lerna run format --stream",
     "postinstall": "husky install",
     "lint": "eslint scripts/*.js && lerna run lint --stream",
     "prepare": "yarn build",
-    "start": "start-storybook -s packages/bedrock/dist -p 6006",
+    "start": "storybook dev -p 6006",
```
Comment (jzempel) [inline package.json:12]: Why were the static directories removed?

File: .storybook/manager-head.html [hunk 11]
```diff
@@ -1 +0,0 @@
-<link rel="icon" href="https://zendeskgarden.github.io/favicons/garden/favicon.ico" />
```
Comment (jzempel) [inline .storybook/manager-head.html:1]: ..and restore

**Review by jzempel — CHANGES_REQUESTED** [review]

File: package.json (hunk 10 above)
Comment (Francois-Esquire) [reply package.json:12]: They weren't, the `-s` (`--static-dirs`) has been deprecated and the configuration has been moved, [see here](https://storybook.js.org/docs/react/api/main-config-static-dirs#page-top).

**Pull request merged**